OpenOversight/app/main/views.py

```diff
@@ -74,26 +80,49 @@ def submit_complaint():
                            officer_image=request.args.get('officer_image'))
 
 
-@main.route('/submit')
+@main.route('/submit', methods=['GET', 'POST'])
+@login_required
 def submit_data():
-    return render_template('submit.html')
-
-
-@main.route('/upload', methods=['POST'])
-def upload_file():
-    file = request.files['file']
-    if file and allowed_file(file.filename):
-        filename = secure_filename(file.filename)
-        file.save(os.path.join(current_app.config['UNLABELLED_UPLOADS'], filename))
-        return redirect(url_for('main.show_upload',
-                                filename=filename))
-
-
-@main.route('/show_upload/<filename>')
-def show_upload(filename):
-    # return send_from_directory('../'+config.UNLABELLED_UPLOADS,
-    #                           filename)
-    return 'Successfully uploaded: {}'.format(filename)
+    form = HumintContribution()
+    if request.method == 'POST' and form.validate_on_submit():
+        original_filename = secure_filename(request.files[form.photo.name].filename)
+        image_data = request.files[form.photo.name].read()
+
+        # See if there is a matching photo already in the db
+        hash_img = compute_hash(image_data)
+        hash_found = Image.query.filter_by(hash_img=hash_img).first()
+
+        if not hash_found:
+            # Generate new filename
+            file_extension = original_filename.split('.')[-1]
+            new_filename = '{}.{}'.format(hash_img, file_extension)
+
+            # Save temporarily on local filesystem
+            tmpdir = tempfile.mkdtemp()
+            safe_local_path = os.path.join(tmpdir, new_filename)
+            with open(safe_local_path, 'w') as tmp:
+                tmp.write(image_data)
+            os.umask(SAVED_UMASK)
+
+            # Upload file from local filesystem to S3 bucket and delete locally
+            url = upload_file(safe_local_path, original_filename, new_filename)
```

Comment:
this line can throw boto exceptions if s3 is down or the credentials are incorrect; do we want to display an informative message to the user when this happens (`flash("Your file could not be uploaded at this time due to a server problem. Please retry again later.")` or the like)? I think if we don't, we'll just show the default nginx unstyled 500.